examples/models/cubicmap.lisp

```diff
@@ -0,0 +1,67 @@
+(in-package #:cl-user)
+(defpackage claylib/examples/cubicmap
+  (:use :cl :claylib)
+  (:export :main))
+(in-package #:claylib/examples/cubicmap)
+
+(defparameter *scene*
+  (make-scene ((image-ass (make-image-asset (claylib/examples:claylib-path
+                                             "examples/models/resources/cubicmap.png")))
+               (texture (make-texture-asset (claylib/examples:claylib-path
+                                             "examples/models/resources/cubicmap_atlas.png"))))
+              ((cubicmap (let ((cubicmap (make-empty-texture
+                                          :rot 0
+                                          :tint +white+)))
+                           (load-texture-from-image (asset image-ass) :texture cubicmap)
+                           (setf (source cubicmap)
+                                 (make-simple-rec 0 0 (width cubicmap) (height cubicmap))
+                                 (dest cubicmap)
+                                 (make-simple-rec (- (get-screen-width) (* 4 (width cubicmap)) 20) 20
+                                                  (* 4 (width cubicmap)) (* 4 (height cubicmap))))
+                           cubicmap))
+               (model (let ((m (load-model-from-mesh (gen-mesh-cubicmap (asset image-ass)
+                                                                        (make-vector3 1 1 1)))))
+                        ;; Change the texture of the diffuse map of 0th material in the model
+                        (set-slot :texture
+                                  (elt (maps (elt (materials m) 0)) +material-map-diffuse+)
+                                  (asset texture))
+
+                        (setf (pos m) (make-vector3 -16 0 -8))
+                        ;; TODO Why doesn't this work?
+                        ;; (setf (x m) -16
+                        ;;       (y m) 0
+                        ;;       (z m) -8)
+                        m))
+               (camera (make-camera-3d 16 14 16
+                                       0 0 0
+                                       0 1 0
+                                       :mode +camera-orbital+))
+               (rec (make-rectangle (- (get-screen-width)
+                                       (* 4 (width (eager-future2:yield cubicmap)))
+                                       20)
+                                    20
+                                    ;; TODO shouldn't have to yield futures as a user
+                                    (* 4 (width (eager-future2:yield cubicmap)))
+                                    (* 4 (height (eager-future2:yield cubicmap)))
```

Comment:
No they shouldn't, but why is it necessary here anyway? `rec` isn't yielded until `with-scenes` just like everything else, right?

Reply:
> `rec` isn't yielded until `with-scenes` just like everything else, right?

Yes, it isn't, or at least it shouldn't be. The issue appears to be that `cubicmap` references a future and *not* the future's value, regardless of whether it's yielded. I am calling yield only as a way to get at the value of `cubicmap`, which should already have been yielded at this point by `set-up-scene`. I'll look into this further because it obviously needs fixing.